### ISSUE TITLE
Speaker detection for intros with multiple people

### ISSUE THREAD

**Issue by rbbby** [welfare-state-analytics/riksdagen-corpus#134]
Strange bug where first intro of protocol is identified as "1977/78:", which in itself we are aware of and can be taken into account by regex. But why is the first speaker identified as Q6216923, Bo Turesson? Found in 197778/prot-197778--5.xml.

<img width="654" alt="Screenshot 2022-02-22 at 20 39 00" src="https://user-images.githubusercontent.com/78153594/155206369-637c97dd-8ad3-4a72-9984-663b73216e97.png">

**Comment by Stubbendorff** [comment]
I have seen that in other cases as well, at least in version 0.3. it is not the speaker (Kondradsson) who is tagged, but the minister (Turesson) who is adressed.

**Comment by rbbby** [comment]
Ah thanks that brings some clarity to it. The algorithm does not take object-subject information into account. Atm it looks at this speech and finds name: Ingemar Konradsson, party: S, other: kommunikationsminister. As it has identified a minister in the speech, it will try and match against the minister database (and does so aggressively at a minimum only requiring that we have a unique kommunikationsminister in the relevant time period). Can try solving it by not identifying "(...)ministern" and similar if there is a "till" before it. Multiple people being mentioned in an intro can cause tricky situations.

**Comment by MansMeg** [comment]
Yes. Good that we isolated this bug. The question is how big this problem is.

**Comment by ninpnin** [comment]
This should be solved with BERT.
